less_users/models.py

```diff
@@ -62,18 +46,39 @@ def get_points(self):
         and this challenge makes possible to get points once for every 7 days;
         for 1/day challenge points can be gained each day"""
         points = self.challenge.points
-        if len(self.log_set.all()) > 0:
-            last_log = int(self.log_set.last().date.strftime("%Y%m%d"))
-        else:
-            last_log = 0
         frequency = self.challenge.frequency
-        today = self.todays_day_num()
-        if today - last_log > frequency:
+        date_today = date.today()
+        self.check_if_active
```

Comment:
Generalnie nie podoba mi się, że funkcja, która coś „sprawdza” jest oznaczona jako property.

Reply:
`property` to coś co jest własnością obiektu, coś stałego, coś co po prostu zwraca jakąś wartość. Ty natomiast w tej funkcji SPRAWDZASZ czy jest aktywna. I do tego ustawiasz czy `is_active`. Dlatego jest to coś co nadaje się na metodę, nie property. 

> teoretycznie mogłabym ustawić getter i setter, ale to tylko dla ukrytych pól w modelu

Getter i setter nie muszą być robione tylko dla ukrytych pól. Plus tutaj to już w ogóle byłby przerost formy nad treścią ;)

Reply:
chodzi Ci o to, żeby zrezygnować z dekoratora @property i zrobić z tej funkcji metodę `get_points`?

Reply:
Nie! Chodzi mi o to, żeby `check_if_active` było metodą.

Reply:
OK, rozpatrywałam komentarz jak należący do funkcji `def get_points`, a nie tylko do linijki gdzie skorzystałam z `check_if_active`. Teraz już rozumiem o co Tobie chodziło. `check_if_active` jest metodą, bo coś oblicza w tle, a nie podaje samą wartość przypisaną do obiektu.